js/timeout-transition-group.jsx

```diff
@@ -87,6 +84,40 @@ function animationSupported() {
     return endEvents.length !== 0;
 }
 
+/**
+ * Functions for element class management to replace dependency on jQuery
+ * addClass, removeClass and hasClass
+ */
+function addClass(element, className) {
+    if (element.classList) {
+        element.classList.add(className);
+    } else if (!hasClass(element, className)) {
+        element.className = element.className + ' ' + className;
+    }
+    return element;
+}
+function removeClass(element, className) {
+    if (hasClass(className)) {
+        if (element.classList) {
+            element.classList.remove(className);
+        } else {
+            element.className = element.className
+                .replace(className, '')
```

Comment:
Okay, I'm still a bit confused. Could I walk through an example and maybe we can figure out what I'm missing?

Let's say `element.className` is `"class1 class2 class3"` and `className` is `"class2"`.

```javascript
element.className.replace(className, '').replace(new RegExp('/(^|\\s)' + className + '($|\\s)/'), ' ')
```
returns `"class1  class3"` (with two spaces, although github markdown rendering doesn't show the second space)

while

```javascript
element.className.replace(new RegExp('(^|\\s)' + className + '($|\\s)'), ' ')
```
returns `"class1 class3"` (I removed the leading and trailing `/` from your regexp--you need those if it's a literal, but shouldn't have them in a RegExp constructor unless you want to match a literal '/').

It seems to me like we can just use the latter replacement here.